drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -794,6 +832,336 @@ local function match_profile(driver, device)
+  end
+  if #rock_eps > 0 then
+    table.insert(main_component_capabilities, capabilities.fanOscillationMode.ID)
+  end
+  if #wind_eps > 0 then
+    table.insert(main_component_capabilities, capabilities.windMode.ID)
+  end
+
+
+  local thermostat_capabilities = get_thermostat_optional_capabilities(device)
+  for _, capability_id in pairs(supported_capabilities) do
+    table.insert(main_component_capabilities, capability_id)
+  end
+
+  local battery_supported = device:get_field(profiling_data.BATTERY_SUPPORT)
+  if battery_supported == battery_support.BATTERY_LEVEL then
+    table.insert(main_component_capabilities, capabilities.batteryLevel.ID)
+  elseif battery_supported == battery_support.BATTERY_PERCENTAGE then
+    table.insert(main_component_capabilities, capabilities.battery.ID)
+  end
+
+  table.insert(optional_supported_component_capabilities, {"main", main_component_capabilities})
+  device:try_update_metadata({profile = profile_name, optional_component_capabilities = optional_supported_component_capabilities})
+
+  -- add mandatory capabilities for subscription
+  local total_supported_capabilities = optional_supported_component_capabilities
+  -- TODO: make sure these are added to the main component list, even though it theoretically shouldn't matter
+  -- however, the numbering is thrown off if there are other components for hepa/AC filter
+  table.insert(main_component_capabilities, capabilites.thermostatMode.ID)
+  table.insert(main_component_capabilities, capabilites.temperatureMeasurement.ID)
+
+  device:set_field(SUPPORTED_COMPONENT_CAPABILITIES, total_supported_capabilities, { persist = true })
+
+  --re-up subscription with new capabiltiies using the moudlar supports_capability override
+  device:extend_device("supports_capability_by_id", supports_capability_by_id_modular)
+  device:subscribe()
+end
+
+local function match_modular_profile(driver, device)
+  if profiling_data_still_required(device) then return end
+
+  if device_type == RAC_DEVICE_TYPE_ID then
+    match_modular_profile_room_ac(driver, device)
+  elseif device_type == FAN_DEVICE_TYPE_ID then
+    match_modular_profile_fan(driver, device)
+  elseif device_type == AP_DEVICE_TYPE_ID then
+    match_modular_profile_air_purifer(driver, device)
+  elseif device_type == WATER_HEATER_DEVICE_TYPE_ID then
+    -- TODO
+  elseif device_type == HEAT_PUMP_DEVICE_TYPE_ID then
+    -- TODO
+  elseif #thermostat_eps > 0 then
+   match_modular_profile_thermostat(driver, device)
+  else
+    device.log.warn_with({hub_logs=true}, "Device type is not supported in thermostat driver")
+    return
+  end
+
+  -- clear all profiling data fields after profiling is complete.
+  for _, field in pairs(profiling_data) do
+    device:set_field(field, nil)
+  end
+end
+
+local function match_profile(driver, device)
+  -- must use profile switching on older hubs
+  if version.api < 14 and version.rpc < 7 then
```

Comment:
I think this should be an or, if either one is too low we use a static profile

Reply:
Good catch - I updated this to an `or` instead, and I also left a note to update the RPC version to 8 once that gets into a build since I bumped the version number for the sake of tracking modular profile support :+1: